solutions/task_9.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+"""
+TASK: Implement following context managers to execute unittests without errors:
+class 'MyHttpCtxManagerClass'
+function 'my_http_ctx_manager_func'
+"""
+import unittest
+from contextlib import contextmanager
+from http.client import HTTPSConnection, HTTPConnection, NotConnected
+
+HTTPConnection.auto_open = False
+
+
+class MyHttpCtxManagerClass:
+
+    def __init__(self, url):
+        self.url = url
+
+    def __enter__(self):
+        self.connection = HTTPSConnection(self.url)
```

Comment:
To define all object attributes inside ```__init__``` is more readable. You will have all attributes in one place. So if your class is large it's easy to view attributes and understand class logic and functionality.

Often all attributes can be defined as ```self.connection = None```, but in this case you can create connection in the ```__init__``` (it is not connected yet, just created).

Reply:
https://stackoverflow.com/questions/19284857/instance-attribute-attribute-name-defined-outside-init